src/components/CommentForm/CommentForm.jsx

```diff
@@ -0,0 +1,98 @@
+import { useParams } from "react-router-dom";
+import { useContext, useState } from "react";
+import "./CommentForm.css";
+import { UserContext } from "../Contexts/User";
+import { postCommentByArticleId } from "../../../utils/api.comments";
+
+const CommentForm = ({
+	setIsCommenting,
+	setCommentsData,
+	commentError,
+	setCommentError,
+}) => {
+	const [input, setInput] = useState("");
+	const { article_id } = useParams();
+	const { user } = useContext(UserContext);
+	const { username } = user;
+	//! const username = "testError";
+
+	const handleToggleComment = () => {
+		setIsCommenting((current) => !current);
+	};
+
+	const handleInput = (e) => {
+		e.preventDefault();
+		setInput(e.target.value);
+		setCommentError(false);
+	};
+
+	const handleSubmitComment = (e) => {
+		e.preventDefault();
+		setCommentError(false);
+
+		postCommentByArticleId(article_id, username, input)
+			.then(() => {
+				handleToggleComment();
+			})
+			.catch((err) => {
+				setCommentError(true);
+
+				setCommentsData((current) => {
+					const { comments, pages } = current;
+					const copiedComments = [...comments].slice(1);
+
+					return {
+						comments: copiedComments,
+						pages,
+					};
+				});
+			});
+
+		setCommentsData((current) => {
```

Comment:
nice optimistic render.  you can use comment from the API, but this is grand also!